### PR TITLE
fuzz: save/load DHCP client leases

### PR DESCRIPTION
**Pull request by evverx**
```
==25679==ERROR: LeakSanitizer: detected memory leaks

Direct leak of 64 byte(s) in 1 object(s) allocated from:
    #0 0x511e2f in __interceptor_realloc (/home/vagrant/systemd/out/fuzz-lease-load+0x511e2f)
    #1 0x7f9588047529 in greedy_realloc /home/vagrant/systemd/build/../src/basic/alloc-util.c:70:13
    #2 0x7f958806e3f7 in parse_env_file_internal /home/vagrant/systemd/build/../src/basic/env-file.c:120:38
    #3 0x7f958806f1d5 in parse_env_filev /home/vagrant/systemd/build/../src/basic/env-file.c:348:13
    #4 0x7f958806f1d5 in parse_env_file_sentinel /home/vagrant/systemd/build/../src/basic/env-file.c:365:13
    #5 0x5530ff in dhcp_lease_load /home/vagrant/systemd/build/../src/libsystemd-network/sd-dhcp-lease.c:1151:13
    #6 0x54a2b1 in LLVMFuzzerTestOneInput /home/vagrant/systemd/build/../src/libsystemd-network/fuzz-lease-load.c:33:16
    #7 0x44a099 in fuzzer::Fuzzer::ExecuteCallback(unsigned char const*, unsigned long) (/home/vagrant/systemd/out/fuzz-lease-load+0x44a099)
    #8 0x44a84b in fuzzer::Fuzzer::MinimizeCrashLoop(std::vector<unsigned char, fuzzer::fuzzer_allocator<unsigned char> > const&) (/home/vagrant/systemd/out/fuzz-lease-load+0x44a84b)
    #9 0x42a1a7 in fuzzer::MinimizeCrashInputInternalStep(fuzzer::Fuzzer*, fuzzer::InputCorpus*) (/home/vagrant/systemd/out/fuzz-lease-load+0x42a1a7)
    #10 0x432753 in fuzzer::FuzzerDriver(int*, char***, int (*)(unsigned char const*, unsigned long)) (/home/vagrant/systemd/out/fuzz-lease-load+0x432753)
    #11 0x421ca6 in main (/home/vagrant/systemd/out/fuzz-lease-load+0x421ca6)
    #12 0x7f95876bb55f in __libc_start_call_main (/lib64/libc.so.6+0x2d55f)

SUMMARY: AddressSanitizer: 64 byte(s) leaked in 1 allocation(s).
INFO: to ignore leaks on libFuzzer side use -detect_leaks=0.
```

### CODE REVIEW

File: src/libsystemd-network/fuzz-lease-load.c [hunk 1]
```diff
@@ -0,0 +1,24 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "dhcp-lease-internal.h"
+#include "fd-util.h"
+#include "fs-util.h"
+#include "fuzz.h"
+#include "network-internal.h"
+#include "tmpfile-util.h"
+
+
```
Comment (yuwata) [inline src/libsystemd-network/fuzz-lease-load.c:10]: Doubled blank line.

File: test/fuzz/fuzz-lease-load/init [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+SIP=
+POP3=
+SMTP=
+LPR=
+MTU=
+DOMAINNAME=
+HOSTNAME=
+DOMAIN_SEARCH_LIST=
+ROOT_PATH=
+ROUTES=
```
Comment (yuwata) [inline test/fuzz/fuzz-lease-load/init:18]: Please change this `STATIC_ROUTES` and `CLASSLESS_ROUTES`. See 32f8a613c542a30eb48317db3f446360afa89a73.

**Review by yuwata — CHANGES_REQUESTED** [review]
Please rebase and re-trigger CIs.

**Comment by evverx** [comment]
@yuwata thanks! I'll change that. I think it would make sense to also call `dhcp_lease_save` to make sure that `networkd` doesn't crash when it serializes its state. It's more important than loading "state" files that can affect networkctl only as far as I can see.

**Comment by yuwata** [comment]
> @yuwata thanks! I'll change that. I think it would make sense to also call `dhcp_lease_save` to make sure that `networkd` doesn't crash when it serializes its state. It's more important than loading "state" files that can affect networkctl only as far as I can see.

Sounds nice.

**Comment by evverx** [comment]
@yuwata I'll try to add it next week probably hopefully along with the fuzzer I mentioned in https://github.com/systemd/systemd/pull/22365.

**Comment by evverx** [comment]
I'm not planning to work on this any more.

**Pull request closed without merging**